sdk/typescript/runtime/main.go

```diff
@@ -535,10 +581,13 @@ func (t *TypescriptSdk) analyzeModuleConfig(ctx context.Context, modSource *dagg
 	}
 
 	// If a first init, there will be no directory, so we ignore the error here.
+	// We also only include package.json & lockfiles to benefit from caching.
 	t.moduleConfig.source = modSource.ContextDirectory().Directory(t.moduleConfig.subPath)
-	entries, err := t.moduleConfig.source.Entries(ctx)
+	configEntries, err := dag.Directory().WithDirectory(".", t.moduleConfig.source, dagger.DirectoryWithDirectoryOpts{
+		Include: []string{"package.json", "*lock*", "tsconfig.json", "pnpm-workspace.yaml", ".yarnrc.yml"},
```

Comment:
Isn't this also too broad? Same as in https://github.com/dagger/dagger/pull/8236#discussion_r1735072725

Reply:
You're right! I'll do the update

Reply:
Fixed in [4236a3f](https://github.com/dagger/dagger/pull/8236/commits/4236a3fa44a24681968da6b4530726cd1aac19ab)